e2e-tests/test/anvil-apis.test.ts

```diff
@@ -0,0 +1,190 @@
+import { expect } from "chai";
+import { Wallet } from "zksync-web3";
+import { deployContract, expectThrowsAsync, getTestProvider } from "../helpers/utils";
+import { RichAccounts } from "../helpers/constants";
+import { ethers } from "hardhat";
+import { Deployer } from "@matterlabs/hardhat-zksync-deploy";
+import * as hre from "hardhat";
+import { keccak256 } from "ethers/lib/utils";
+import { BigNumber } from "ethers";
+
+const provider = getTestProvider();
+
+describe("anvil_setBalance", function () {
+  it("Should update the balance of an account", async function () {
+    // Arrange
+    const userWallet = Wallet.createRandom().connect(provider);
+    const newBalance = ethers.utils.parseEther("42");
+
+    // Act
+    await provider.send("anvil_setBalance", [userWallet.address, newBalance._hex]);
+
+    // Assert
+    const balance = await userWallet.getBalance();
+    expect(balance.eq(newBalance)).to.true;
+  });
+});
+
+describe("anvil_setNonce", function () {
+  it("Should update the nonce of an account", async function () {
+    // Arrange
+    const userWallet = Wallet.createRandom().connect(provider);
+    const newNonce = 42;
+
+    // Act
+    await provider.send("anvil_setNonce", [userWallet.address, ethers.utils.hexlify(newNonce)]);
+
+    // Assert
+    const nonce = await userWallet.getNonce();
+    expect(nonce).to.equal(newNonce);
+  });
+});
+
+describe("anvil_mine", function () {
+  it("Should mine multiple blocks with a given interval", async function () {
+    // Arrange
+    const numberOfBlocks = 100;
+    const intervalInSeconds = 60;
+    const startingBlock = await provider.getBlock("latest");
+    const startingTimestamp: number = await provider.send("config_getCurrentTimestamp", []);
+
+    // Act
+    await provider.send("anvil_mine", [ethers.utils.hexlify(numberOfBlocks), ethers.utils.hexlify(intervalInSeconds)]);
+
+    // Assert
+    const latestBlock = await provider.getBlock("latest");
+    expect(latestBlock.number).to.equal(startingBlock.number + numberOfBlocks, "Block number mismatch");
+    expect(latestBlock.timestamp).to.equal(
+      startingTimestamp + (numberOfBlocks - 1) * intervalInSeconds * 1000 + 1,
+      "Timestamp mismatch"
+    );
+  });
+});
+
+describe("anvil_impersonateAccount & anvil_stopImpersonatingAccount", function () {
+  it("Should allow transfers of funds without knowing the Private Key", async function () {
+    // Arrange
+    const userWallet = Wallet.createRandom().connect(provider);
+    const beforeBalance = await provider.getBalance(RichAccounts[5].Account);
```

Comment:
nit: `RichAccounts[5].Account` repeats everywhere . Create a sep var for this

Reply:
OK, name added.